README.md

```diff
@@ -1,636 +1,524 @@
+`mysql/terraform.tfvars`, but this approach is much easier to maintain! Note that if you include any other settings
+in the `terragrunt` block of a child `.tfvars` file, it will override the settings in the parent.
 
-```bash
-ENVIRONMENT_VARIABLE="value" terragrunt
-```
+The `terraform.tfvars` files above use two *helper functions*: 
 
-Then the previous example would evaluate to:
+* `find_in_parent_folders()`: This helper returns the path to the first `terraform.tfvars` file it finds in the parent 
+  folders above the current `terraform.tfvars` file. In the example above, the call to `find_in_parent_folders()` in 
+  `mysql/terraform.tfvars` will return `../terraform.tfvars`. This way, you don't have to hard code the `path` 
+  parameter in every module.
 
-```hcl
-terragrunt = {
-  remote_state {
-    backend = "s3"
-    config {
-      encrypt = "true"
-      bucket = "value-bucket"
-      key = "/foo/bar/terraform.tfstate"
-      region = "us-west-2"
-    }
-  }
-}
-```
+* `path_relative_to_include()`: This helper function returns the relative path between the current `terraform.tfvars` 
+  file and the path specified in its `include` block. We use this so that each Terraform module stores its Terraform
+  state at a different `key`. For example, the `mysql` module will have its `key` parameter resolve to 
+  `mysql/terraform.tstate` and the `frontend-app` module will have its `key` parameter resolve to 
+  `frontend-app/terraform.tfstate`.
 
-Terraform itself also supports loading variables via the environment. It is possible to use the same variables by 
-correctly using the terraform prefix `TF_VAR_`.
+See [the helper functions docs](#helper-functions) for more info.
 
-```bash
-TF_VAR_variable="value" terragrunt apply
-```
+
+#### Create remote state and locking resources automatically
+
+When you run `terragrunt` with `remote_state` configuration, it will automatically create the following resources if 
+they don't already exist:
+
+* **S3 bucket**: If you are using the [S3 backend](https://www.terraform.io/docs/backends/types/s3.html) for remote 
+  state storage and the `bucket` you specify in `remote_state.config` doesn't already exist, Terragrunt will create it 
+  automatically, with [versioning enabled](http://docs.aws.amazon.com/AmazonS3/latest/dev/Versioning.html). 
+
+* **DynamoDB table**: If you are using the [S3 backend](https://www.terraform.io/docs/backends/types/s3.html) for 
+  remote state storage and you specify a `lock_table` (a [DynamoDB table used for 
+  locking](https://www.terraform.io/docs/backends/types/s3.html#lock_table)) in `remote_state.config`, if that table
+  doesn't already exist, Terragrunt will create it automatically, including a primary key called `LockID`.
+
+**Note*: If you specify a `profile` key in `remote_state.config`, Terragrunt will automatically use this AWS profile 
+when creating the S3 bucket or DynamoDB table. 
```

Comment:
Consider the use case of adding Route53 Public Hosted Zones. So your position would be to write a trivial Terraform module that does the following:

```hcl
resource "aws_route53_zone" "primary" {
  name = "${var.domain_name}"
}
```

...so that you could have the following Terragrunt files in the `live` repo:

```hcl
terragrunt = {
  include {
    path = "${find_in_parent_folders()}"
  }
  terraform {
    source = "git::git@github.com:foo/modules.git//route53_hosted_zone?ref=v0.0.3"
  }
}

domain_name = "example.com"
# Alternatively
domain_names = ["example.com"]
```

I guess you're arguing the standardization gives us the `apply-all` command, and forces any complexity into a versionable module. In this extreme trivial case, the overhead outweighs the value but still gives us consistency.

I'd say the one potential area is this may break down a little is if Terraform's HCL isn't expressive enough to capture what we want. For example, I often wish modules accepted lists of lists, but since we can only pass "flat" lists, we may have to instantiate one module per each "thing".

Mostly thinking out loud here...